components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,331 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCActionSheetController.h"
+#import "MDCActionSheetItemView.h"
+#import "MDCActionSheetListViewController.h"
+#import "MaterialBottomSheet.h"
+#import "MaterialApplication.h"
+#import "MaterialTypography.h"
+
+//NSString *const kTableReuseIdentifier = @"BaseCell";
+
+@interface MDCActionSheetAction ()
+
+@property(nonatomic, nullable, copy) MDCActionSheetHandler completionHandler;
+
+@end
+
+@implementation MDCActionSheetAction
+
++ (instancetype)actionWithTitle:(nonnull NSString *)title
+                          image:(nullable UIImage *)image
+                          handler:(void (^__nullable)(MDCActionSheetAction *action))handler {
+    return [[MDCActionSheetAction alloc] initWithTitle:title
+                                                 image:image
+                                               handler:handler];
+}
+
+- (instancetype)initWithTitle:(nonnull NSString *)title
+                        image:(nullable UIImage *)image
+                      handler:(void (^__nullable)(MDCActionSheetAction *action))handler {
+  self = [super init];
+  if (self) {
+    _title = [title copy];
+    _image = [image copy];
+    _completionHandler = [handler copy];
+  }
+  return self;
+}
+
+-(id)copyWithZone:(__unused NSZone *)zone {
+  MDCActionSheetAction *action = [[self class] actionWithTitle:self.title
+                                                         image:self.image
+                                                       handler:self.completionHandler];
+  return action;
+}
+
+@end
+
+@interface MDCActionSheetController () <MDCBottomSheetPresentationControllerDelegate, UITableViewDelegate>
+
+@property(nonatomic, nonnull) MDCActionSheetHeaderView *header;
+
+@property(nonatomic, nullable) MDCActionSheetListViewController *tableView;
```

Comment:
You have a couple private properties but mostly ivars. Consider making these ivars to make the code more consistent and to keep your private instance properties/values together.

Reply:
Done